### PR TITLE
Azcopy sync option for file share is not supported

### PR DESCRIPTION
**Pull request by dphulkar-msft**
**Issue Link:** https://github.com/Azure/azure-storage-azcopy/issues/2650

**Issue:**
The customer found the data loss issues when they used "Azcopy sync" to sync their on-premises files (millions of files) to Azure files. They detected few thousands of files are corrupted. 
Azcopy sync operation is not supported and is already documented [here](https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-files#synchronize-files) 

**Resolution:**
Added a warning message which will be displayed on the console and in the job logs as well if the user tries to perform sync operation for file share.

**Console Output:**
Screenshot 2024-05-02 163228

**Job Log Output:**
[32b9249b-c4d7-9947-4352-1d7011f7eba9.log](https://github.com/Azure/azure-storage-azcopy/files/15187272/32b9249b-c4d7-9947-4352-1d7011f7eba9.log)

### CODE REVIEW

File: cmd/zt_sync_local_blob_test.go [hunk 1]
```diff
@@ -468,3 +469,53 @@ func TestDryrunSyncLocaltoBlob(t *testing.T) {
 		a.True(testDryrunStatements(blobsToDelete, msg))
 	})
 }
+
+// TestSyncGenerateWarningForLocalToFSUpload
```
[the remark this answers is not in the document]
Comment (dphulkar-msft) [reply cmd/zt_sync_local_blob_test.go:473]: sure. Should I create a new test file for this or do you want me to add to any existing test file? If existing, can you point me to the appropriate test file?

**Review by pranavmalik-msft — CHANGES_REQUESTED** [review]
Have we disabled the E2E test-cases? I can observe that they are not running in your case.

**Comment by adreed-msft** [comment]
> Have we disabled the E2E test-cases? I can observe that they are not running in your case.

Seems to be caused by not originating from within the repo.

**Comment by gapra-msft** [comment]
Closing this PR as this is a duplicate

**Pull request closed without merging**